### PR TITLE
doc for x-extensions

### PR DESCRIPTION
**Pull request by pamgoodrich**
See Slack thread:
https://stoplight-internal.slack.com/archives/C0258SZH9SB/p1659563375494659?thread_ts=1659561667.201409&cid=C0258SZH9SB

### CODE REVIEW

File: 3.-design/x-extensions.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  }
+```
+
+### OAS 2.x Support
+
+This set of tags is used to extend support  OAS 2.x documents: 
+
+- x-deprecated
+- x-example
+- x-examples
```
Comment (P0lip) [inline 3.-design/x-extensions.md:54]: This one is also used for OAS 3.0

File: 3.-design/x-extensions.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+- x-nullable
+
+### Migration Support
+
+This set of tags is used to ensure smooth transition to Stoplight from other tools:
+
+- x-logo
+- x-tags
```
Comment (P0lip) [inline 3.-design/x-extensions.md:62]: This one is for our own usage - we use it for tags in JSON Schema, because, unlike OpenAPI, JSON Schema doesn't support them natively.

File: 3.-design/x-extensions.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  "x-stoplight": {
+    "id": "jknwvq9tvtozp"
+  }
+```
+
+### OAS 2.x Support
+
+This set of tags is used to extend support  OAS 2.x documents: 
+
+- x-deprecated
```
Comment (P0lip) [inline 3.-design/x-extensions.md:52]: This one is only used for models in OAS 2.x documents in case that matters.
We use `deprecated` for operation objects.

File: 3.-design/x-extensions.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+```json
+  "x-stoplight": {
+    "id": "jknwvq9tvtozp"
+  }
+```
+
+### OAS 2.x Support
+
+This set of tags is used to extend support  OAS 2.x documents:
```
Comment (P0lip) [inline 3.-design/x-extensions.md:50]:
```suggestion
This set of tags is used to extend support OAS 2.x documents: 
```

File: 3.-design/x-extensions.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+Extensions are used to add capabilities to an API beyond what is provided by a base specification, such as Open API. 
+
+Extensions:
+
+- Are always prefixed with `x-`.
+- Do not appear in published documentation.
+
+You can use the **Form Editor** to add extensions to these areas of your API document:
+
+- Info object
```
Comment (P0lip) [inline 3.-design/x-extensions.md:16]: The only specification extension we support inside of the Info object is `x-logo`, but we don't expose that.
We add `x-internal` to OpenAPI object though, so we could rephrase it to
```suggestion
- OpenAPI object
```

File: 3.-design/x-extensions.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+- Operations object
```
[the remark this answers is not in the document]
Comment (pamgoodrich) [reply 3.-design/x-extensions.md:17]: Done

File: 3.-design/x-extensions.md (hunk 1 above)
Comment (pamgoodrich) [reply 3.-design/x-extensions.md:54]: - x-examples (also supported in OAS 3.x documents)

File: 3.-design/x-extensions.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+This set of tags is used to extend support  OAS 2.x documents: 
+
+- x-deprecated
+- x-example
+- x-examples
+- x-nullable
+
+### Migration Support
+
+This set of tags is used to ensure smooth transition to Stoplight from other tools:
```
[the remark this answers is not in the document]
Comment (pamgoodrich) [reply 3.-design/x-extensions.md:59]: I most certainly did! Thank you!

File: 3.-design/x-extensions.md (hunk 4 above)
Comment (pamgoodrich) [reply 3.-design/x-extensions.md:50]: done

**Review by P0lip — APPROVED** [review]
Apart from https://github.com/stoplightio/platform-docs/pull/515/files#r966040638, it looks good to me, so I'm giving it a preemptive approval.

File: 3.-design/x-extensions.md (hunk 2 above)
Comment (pamgoodrich) [reply 3.-design/x-extensions.md]: Oops. Missed that one. Just pushed the change.

**Review by P0lip — APPROVED** [review]

**Review by heitortsergent — APPROVED** [review]
Just left a few comments, looks great!

One question I have is, we don't have the term "x-extensions" in the new page, so I was wondering if it might be good to add it somewhere there so it shows up in the search?

**Comment by pamgoodrich** [comment]
> Just left a few comments, looks great!
> 
> One question I have is, we don't have the term "x-extensions" in the new page, so I was wondering if it might be good to add it somewhere there so it shows up in the search?

Good catch! Thank you.

> 
> Also, I'm wondering if there's a bug with how search works if you're viewing a specific branch? Because if I go to your branch (https://meta.stoplight.io/docs/platform/branches/pam-446-more-jse-updates/52ab0a117eadd-welcome-to-the-stoplight-docs), and then type in the search bar, and select a result, it redirects me to the main branch. If I try to open the new Extensions page, it return a 404 because it doesn't exist on the main branch...

This is a known limitation and perhaps something we should add to our wish list. There is at least one customer request for this. https://stoplight-internal.slack.com/archives/C0350RJLX1N/p1645628796268909
